### PR TITLE
Harmony 1505/6 - Add extendDimensions parameter to DataOperation

### PR DESCRIPTION
**Pull request by vinnyinverso**
## Jira Issue ID
HARMONY-1505/6

## Description
Adds extendDimensions to the DataOperation. Holding off on API documentation until we are fully operational with the first extend-capable service.

## Local Test Steps
- Under `capabilities:` in services.yml, for harmony-service-example in UAT, add `extend: true`.
- Request: http://localhost:3000/C1233800302-EEDTEST/ogc-api-coverages/1.0.0/collections/all/coverage/rangeset?granuleId=G1233800343-EEDTEST&format=image/tiff&extend=a,b
- Check the workflow_steps table, operation column to ensure the values `a,b` were passed via the extendDimensions property in the operation JSON

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [ ] Documentation updated (if needed)

### CODE REVIEW

File: app/models/data-operation.ts [hunk 1]
```diff
@@ -635,10 +644,28 @@ export default class DataOperation {
...
+   */
+  set extendDimensions(dimensions: string[]) {
+    this.model.extendDimensions = dimensions;
+  }
+
   /**
    * Gets the dimensions used for dimension subsetting
    *
-   * @returns A URI to the geojson shape
+   * @returns The dimensions against which to subset
```
Comment (chris-durbin) [inline app/models/data-operation.ts:668]: Thanks I'm sure that copy/paste was me.

File: app/schemas/data-operation/README.md [hunk 2]
```diff
@@ -20,7 +20,7 @@
 - [ ] Update [harmony-service-lib-py/example/example_message.json](../../../../harmony-service-lib-py/example/example_message.json) to use the new schema
 - [ ] Update [harmony-service-example/example/harmony-operation.json](../../../../harmony-service-example/example/harmony-operation.json) to use the new schema
 - [ ] Update [config/services.yml](../../../config/services.yml) to supply the new library version for any services that will be automatically or manually upgraded with the change
-- [ ] Update [docs/adapting-new-services.md](../../../docs/adapting-new-services.md) to specify the updated version
+- [ ] Update [docs/adapting-new-services.md](../../../docs/guides/adapting-new-services.md) to specify the updated version
 - [ ] Update any Harmony-owned services that need to use the new features or may be incompatible with the change
 - [ ] Notify service authors that the schema was updated
 - [ ] Swear that next time around you'll make the process a little easier
```
Comment (chris-durbin) [inline app/schemas/data-operation/README.md:26]: You forgot this step.

File: config/services.yml [hunk 3]
```diff
@@ -27,7 +27,7 @@ https://cmr.earthdata.nasa.gov:
     description: |
       A service to compose the Giovanni URL and invoke Giovanni service to produce output file to visualize, analyze,
       and access vast amounts of Earth science remote sensing data without having to download the data.
-    data_operation_version: '0.17.0'
+    data_operation_version: '0.18.0'
```
Comment (chris-durbin) [inline config/services.yml:30]: What do you all think about adding a variable at the top of the file for the latest schema version and then do the YAML equivalent of setting LATEST_SCHEMA_VERSION... that way we only update one place when we change the schema, and we make it more obvious if a service is pinned to an older version?

**Review by chris-durbin — COMMENTED** [review]
Looks good - I'll test it out tomorrow morning before approving.

**Review by chris-durbin — APPROVED** [review]
Tested out successfully.

File: config/services.yml (hunk 3 above)
Comment (vinnyinverso) [reply config/services.yml:30]: Sounds good to me if there's a straightforward way to do it.

**Pull request merged**